### PR TITLE
Automated cherry pick of #121919: kube-proxy: Fix etp:Local for externalIPs

### PR DESCRIPTION
**Pull request by uablrek**
Cherry pick of #121919 on release-1.28.

#121919: kube-proxy: Fix etp:Local for externalIPs

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Fixes a 1.26.0+ regression in kube-proxy in ipvs mode to make externalIPs work with externalTrafficPolicy: Local
```

### CODE REVIEW

**Comment by aojea** [comment]
you can undraft, the master PR merged

/lgtm
/approve
/kind bug

**Comment by uablrek** [comment]
@aojea The [cherry pick review](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#cherry-pick-review) requires a `cherry-pick-approved`. Is there anything I should do to initiate that? Or just wait?

I assigned you also to the [cherry-pick for v1.29](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#cherry-pick-review), since you know the problem already. Hope you don't mind.

**Comment by aojea** [comment]
> @aojea The [cherry pick review](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#cherry-pick-review) requires a `cherry-pick-approved`. Is there anything I should do to initiate that? Or just wait?

This is for @kubernetes/release-managers to approve, the backport need technical approval and a second approval from release managers

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
